cmd/ipfs/main.go

```diff
@@ -293,17 +294,22 @@ func makeExecutor(req *cmds.Request, env interface{}) (cmds.Executor, error) {
 	return cmdhttp.NewClient(host, opts...), nil
 }
 
-func getRepoPath(req *cmds.Request) (string, error) {
-	repoOpt, found := req.Options["config"].(string)
-	if found && repoOpt != "" {
-		return repoOpt, nil
-	}
-
+func getRepoAndConfigPath(req *cmds.Request) (string, string, error) {
 	repoPath, err := fsrepo.BestKnownPath()
 	if err != nil {
-		return "", err
+		return "","", err
+	}
+	configPath, found := req.Options["config"].(string)
+	if found && configPath != "" {
+		split := strings.Split(configPath,"/")
+		config.DefaultConfigFile = strings.Join(split[len(split)-1:]," ") //considering the last split as the config file name
+		configPath = ""
+		for i := 0; i<len(split)-1; i++ {
+		    configPath = configPath+split[i]+"/"
+		}
+		return repoPath,configPath, nil
```

Comment:
Seems like using `path.Dir` and `path.Base` would simplify the code here:
```go
config.DefaultConfigFile = path.Base(configPath)
configPath = path.Dir(configPath)
if configPath == "." {
    configPath = ""
}
return repoPath, configPath, nil
```

However, this logic should be in `go-ipfs-config` so that applications other than go-ipfs can access the config file and the repo using the same logic.  I think this can be accomplished by modifying `FilePath` to take a separate `configFilePath` argument, and if it is not set then it does what it currently does.  If a `configFilePath` is given, then it uses that path to the config file. If only a file name is given the the result is `Path(configroot, configFilePath)`.
https://github.com/ipfs/go-ipfs-config/blob/master/config.go#L80-L82